### PR TITLE
feat(PE-7107): update getRecords to return as flat array of objects

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

**Review by dtfiedler — CHANGES_REQUESTED** [review]

File: tests/e2e/esm/index.test.js [hunk 1]
```diff
@@ -544,7 +544,8 @@ describe('e2e esm tests', async () => {
   });
 
   describe('ANT', async () => {
-    const processId = 'YcxE5IbqZYK72H64ELoysxiJ-0wb36deYPv55wgl8xo';
+    // ANT v8 source
+    const processId = 'oQ4GNTed8cnNw-H5olq606gCFd5MbGSV4NTmfpIW4FI';
```
Comment (dtfiedler) [inline tests/e2e/esm/index.test.js:548]: nit: let's add a test for an older ant as well - can be in just a single `it` block

Comment (atticusofsparta) [reply tests/e2e/esm/index.test.js:548]: [0489cb6](https://github.com/ar-io/ar-io-sdk/pull/259/commits/0489cb6924b68c19b8791cc47e0f7c1589be842f) - just a single getInfo call

Comment (dtfiedler) [reply tests/e2e/esm/index.test.js:548]: it should be validating the getRecords api

File: tests/e2e/esm/index.test.js [hunk 2]
```diff
@@ -544,14 +544,27 @@ describe('e2e esm tests', async () => {
...
     const ant = ANT.init({
       process: new AOProcess({
         processId,
         ao: aoClient,
       }),
     });
 
+    it('should be able to get info on old ant', async () => {
```
Comment (dtfiedler) [inline tests/e2e/esm/index.test.js:556]: we changed the getRecords API - we want to test that the SDK correctly maps an old ANTs record response...nothing has changed about the info handler - so this isn't validating anything

Comment (atticusofsparta) [reply tests/e2e/esm/index.test.js:556]: [77601b2](https://github.com/ar-io/ar-io-sdk/pull/259/commits/77601b2fc9117a2a4b9ff1226fa3f329e8a74e1d)

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**